src/components/App.tsx

```diff
@@ -1,16 +1,31 @@
+import { lazy, Suspense } from 'react';
 import { BrowserRouter as Router, Route, Routes } from 'react-router-dom';
 
-import { Index } from '../pages/Index';
-import { FounderSale } from '../pages/FounderSale';
-import { NotFoundPage } from '../pages/NotFoundPage';
+// Lazy loading with preloading
+const Index = lazy(() => import('../pages/Index'));
+const FounderSale = lazy(() => import('../pages/FounderSale'));
+const NotFoundPage = lazy(() => import('../pages/NotFoundPage'));
+
+// Preloading critical routes
+if ('requestIdleCallback' in window) {
```

Comment:
You should not need this import logic, it defeats the purpose of lazy loading, I'm guessing there's a reason why did added it?

I think a better solution starts with data Routers and the lazy loading version, 

https://reactrouter.com/6.28.0/route/lazy